joblib/test/test_parallel.py

```diff
@@ -173,6 +184,23 @@ def g(x):
                             )
 
 
+def test_parallel_timeout_success():
+    # Check that timeout isn't thrown when function is fast enough
+    test_lst = [x ** 2 for x in range(10)]
+    nose.tools.assert_equal(
+               test_lst,
+               Parallel(n_jobs=2, timeout=3)(delayed(fast_f)(x) for x in range(10))
+              )
+
+
+def test_parallel_timeout_fail():
+    # Check that timeout properly fails when function is too slow
+    nose.tools.assert_raises(TimeoutError,
+                             Parallel(n_jobs=2, timeout=5),
+                             (delayed(slow_f)(x) for x in range(10))
+                            )
```

Comment:
Please make the tests run as fast as possible (while still testing the behavior you are interested in:

```python
nose.tools.assert_raises(TimeoutError,
    Parallel(n_jobs=2, timeout=0.01),
    (delayed(sleep)(10) for x in range(10))
 )
```